slang.h

```diff
@@ -344,6 +344,91 @@ convention for interface methods.
 
 #endif // __cplusplus
 
+/* Macros for detecting processor */
```

Comment:
At some point we probably need to move all of the platform-detection stuff out to a companion `slang-platform.h` or something like that, because at this point somebody has to wade through a *bunch* of this preprocessor cruft before they start seeing the actual header.

I'm not too worried about it right now because our actual header is a bit of a mess, but if/when we get around to doing a revamped API with a fresh from-scratch header it might be time to refactor a bit.

Reply:
Yes this would be a good thing to do.